dask_cuda/cuda_worker.py

```diff
@@ -200,53 +200,104 @@ def del_pid_file():
                 },
             )
 
-        self.nannies = [
-            Nanny(
-                scheduler,
-                scheduler_file=scheduler_file,
-                nthreads=nthreads,
-                dashboard=dashboard,
-                dashboard_address=dashboard_address,
-                http_prefix=dashboard_prefix,
-                loop=loop,
-                resources=resources,
-                memory_limit=memory_limit,
-                interface=interface,
-                host=host,
-                preload=(list(preload) or []) + ["dask_cuda.initialize"],
-                preload_argv=(list(preload_argv) or []) + ["--create-cuda-context"],
-                security=security,
-                env={"CUDA_VISIBLE_DEVICES": cuda_visible_devices(i)},
-                plugins={
-                    CPUAffinity(
-                        get_cpu_affinity(nvml_device_index(i, cuda_visible_devices(i)))
-                    ),
-                    RMMSetup(
-                        rmm_pool_size,
-                        rmm_maximum_pool_size,
-                        rmm_managed_memory,
-                        rmm_async,
-                        rmm_log_directory,
-                        rmm_track_allocations,
-                    ),
-                    PreImport(pre_import),
-                },
-                name=name if nprocs == 1 or name is None else str(name) + "-" + str(i),
-                local_directory=local_directory,
-                config={
-                    "distributed.comm.ucx": get_ucx_config(
-                        enable_tcp_over_ucx=enable_tcp_over_ucx,
-                        enable_infiniband=enable_infiniband,
-                        enable_nvlink=enable_nvlink,
-                        enable_rdmacm=enable_rdmacm,
-                    )
-                },
-                data=data(nvml_device_index(i, cuda_visible_devices(i))),
-                worker_class=worker_class,
-                **kwargs,
-            )
-            for i in range(nprocs)
-        ]
+        if nanny:
+            t = Nanny
+            kwargs = {"worker_port": None, "listen_address": None, **kwargs}
```

Comment:
It isn't immediately clear to me, why is this line necessary?

Reply:
I think the `worker_class` arg is something not supported in the `Worker` object.  I'll look again since it's been awhile and see if I can refactor as you are both suggesting.

We should probably do something closer to the following:
https://github.com/dask/distributed/blob/344868ace6cd10f7b99e4cd2d3830fba66a2cdcf/distributed/cli/dask_worker.py#L427-L433

As for `worker_port` and `listen_address` these are both supported by dask-workers but currently not fully support in dask-cuda-workers

Reply:
> I think the `worker_class` arg is something not supported in the `Worker` object.

Right, but here it is being passed to `Nanny`, not `Worker`.

> I'll look again since it's been awhile and see if I can refactor as you are both suggesting.
> 
> We should probably do something closer to the following: https://github.com/dask/distributed/blob/344868ace6cd10f7b99e4cd2d3830fba66a2cdcf/distributed/cli/dask_worker.py#L427-L433

Agreed, this is similar to what I suggested in https://github.com/rapidsai/dask-cuda/pull/834/files#r896411094 .

> As for `worker_port` and `listen_address` these are both supported by dask-workers but currently not fully support in dask-cuda-workers

And why do we need to specify them? Is this something we should add as arguments instead?